drivers/Watchdog.cpp

```diff
@@ -0,0 +1,67 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifdef DEVICE_WATCHDOG
+
+#include "Watchdog.h"
+
+namespace mbed {
+
+watchdog_status_t Watchdog::start(const uint32_t timeout)
+{
+    if (timeout == 0) {
+        return WATCHDOG_STATUS_INVALID_ARGUMENT;
```

Comment:
Can we use MBED_ERROR1() here, there is an error code defined for INVALID_ARGUMENT already by mbed-os error coding - https://github.com/ARMmbed/mbed-os/blob/master/platform/mbed_error.h#L721